src/bf_sal/linux_usr/bf_sys_dma_hugepages.c

```diff
@@ -181,11 +181,17 @@ static uint32_t conv_to_next_aligned(uint32_t val, uint32_t alignment) {
 static void *alloc_huge_pages(size_t size, unsigned int header_offset) {
   size_t actual_size;
   char *ptr;
-
   actual_size = ALIGN_TO_BF_PAGE_SIZE(size + header_offset);
   ptr = (char *)mmap(NULL, actual_size, PROT_READ | PROT_WRITE,
-                     MAP_SHARED | MAP_ANONYMOUS | MAP_POPULATE | MAP_HUGETLB,
+                     MAP_SHARED | MAP_ANONYMOUS | MAP_POPULATE | MAP_HUGETLB |
+                         (21 << MAP_HUGE_SHIFT),
```

Comment:
MAP_HUGE_2MB may be more clear:
```suggestion
                         MAP_HUGE_2MB,
```

Reply:
I tried it but it doesn't compile on my system:
`/root/p4factory2/submodules/target-syslibs/src/bf_sal/linux_usr/bf_sys_dma_hugepages.c: In function â€˜alloc_huge_pagesâ€™:
/root/p4factory2/submodules/target-syslibs/src/bf_sal/linux_usr/bf_sys_dma_hugepages.c:187:26: error: â€˜MAP_HUGE_2MBâ€™ undeclared (first use in this function); did you mean â€˜MAP_HUGETLBâ€™?
  187 |                          MAP_HUGE_2MB,
      |                          ^~~~~~~~~~~~
      |                          MAP_HUGETLB
`